specs/eip4844/p2p-interface.md

```diff
@@ -65,34 +78,35 @@ The new topics along with the type of the `data` field of a gossipsub message ar
+| `blob_sidecar_{index}` | `SignedBlobSidecar` (new) |
 
 #### Global topics
 
-EIP-4844 introduces a new global topic for beacon block and blobs-sidecars.
+EIP-4844 introduces new global topics for blob sidecars.
 
 ##### `beacon_block`
 
-This topic is deprecated and clients **MUST NOT** expose in their topic set to any peer. Implementers do not need to do
-anything beyond simply skip implementation, and it is explicitly called out as it is a departure from previous versioning
-of this topic.
+The *type* of the payload of this topic changes to the (modified) `SignedBeaconBlock` found in EIP4844.
 
-Refer to [the section below](#transitioning-the-gossip) for details on how to transition the gossip.
+##### `blob_sidecar_{index}`
 
-##### `beacon_block_and_blobs_sidecar`
+This topic is used to propagate signed blob sidecars, one for each sidecar index.
 
-This topic is used to propagate new signed and coupled beacon blocks and blobs sidecars to all nodes on the networks.
+The following validations MUST pass before forwarding the `sidecar` on the network, assuming the alias `sidecar = signed_blob_sidecar.message`:
 
-In addition to the gossip validations for the `beacon_block` topic from prior specifications, the following validations MUST pass before forwarding the `signed_beacon_block_and_blobs_sidecar` on the network.
-Alias `signed_beacon_block = signed_beacon_block_and_blobs_sidecar.beacon_block`, `block = signed_beacon_block.message`, `execution_payload = block.body.execution_payload`.
-- _[REJECT]_ The KZG commitments correspond to the versioned hashes in the transactions list
-  -- i.e. `verify_kzg_commitments_against_transactions(block.body.execution_payload.transactions, block.body.blob_kzg_commitments)`
-
-Alias `sidecar = signed_beacon_block_and_blobs_sidecar.blobs_sidecar`.
-- _[IGNORE]_ the `sidecar.beacon_block_slot` is for the current slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance)
-  -- i.e. `sidecar.beacon_block_slot == block.slot`.
-- _[REJECT]_ The KZG commitments in the block are valid against the provided blobs sidecar
-  -- i.e. `validate_blobs_sidecar(block.slot, hash_tree_root(block), block.body.blob_kzg_commitments, sidecar)`
+- _[REJECT]_ The sidecar is for the correct topic --
+  i.e. `sidecar.index` matches the topic `{index}`.
+- _[IGNORE]_ The sidecar is not from a future slot (with a `MAXIMUM_GOSSIP_CLOCK_DISPARITY` allowance)
+- _[IGNORE]_ The sidecar is from a slot greater than the latest finalized slot --
+  i.e. validate that `sidecar.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)`
+- _[REJECT]_ The proposer signature, `signed_blob_sidecar.signature`, is valid with respect to the `sidecar.proposer_index` pubkey.
+- _[IGNORE]_ The sidecar is the only sidecar with valid signature received for the tuple `(sidecar.slot, sidecar.proposer_index, sidecar.index)`.
+  -- Clients MUST discard blocks where multiple sidecars for the same proposer and index have been observed.
```

Comment:
I don't agree with this. This can cause network fragmentation and at the very least should not be buried in a blob gossip condition.

A sophisticated enough of a network adversary could create a permanent split (if "discard" is forever). Imagine partition A that sees block_a and extra blobs, then drops. Whereas partition B sees the same for block_b. But then partion A sees block_b and *not* the extra blobs. They'll accept block_b whereas B rejected it.

It's dangerous to create any of these permanent rejections that have to do with your transient view.

Normally gossip `IGNORE` on duplicates doesn't really create a permanent split due to IWANT/IHAVE and getting reference messages (e.g. the block) and downloading the dependencies (e.g. it;s parent). This is the case with all of our seen flag conditions, it increases the propagation time and reduces amplification factor (messages are gotten moreso by direct req/resp), while reducing the chance a messages becomes canonical.

I think the most we can rely on is that (1) the actor is bound to the proposer and (2) the proposer reduces the chances of their message becoming canonical by spamming bad blobs, thus opportunity cost not to spam.

Reply:
I too am skeptical of this condition, it implies we'd have to cancel block processing or invalidate previously imported blocks if we see an second blob from the proposer after we've seen a valid blob + block combination

Reply:
Right - this needs a different formulation (it was intended to allow two blobs per tuple, but I realize even that is risky) - the core idea here was to allow / encourage the next proposer to drop blocks whose proposer has been spamming the network, similar to how lighthouse penalises late blocks by reorging them.

We can achieve this by changing this "seen" condition to allow N blobs per `[slot, proposer, index]` - the key point is to create conditions favorable for the next proposer to observe all these blobs and therefore not build on them.

If we allow just one blob, it's possible for the network to become saturated with an invalid blob while still having a valid blob reach the proposer - we will then indeed have to fall back on req/resp to "recover" - IWANT/IHAVE will not be enough since ignored messages are not kept in gossipsub caches. The larger the N, the larger the cost but also the greater the chance that any >=2 blobs reach next proposer. Some N is needed, else proposer can keep spamming.

It's somewhat ugly, but there exists an alternate way to solve this dilemma: we keep the seen-once condition as stated here, however if we see a blob whose signature is valid but commitment is not, we try to recover it with req/resp: if we get one, we reset the "seen" condition for the tuple and rebroadcast the "correct" block.

The above will mean that:
* we get the invalid blob with valid signature and send it to our mesh neighbours
* if we get a valid blob, we'll IGNORE
* we notice the blob was invalid later in the pipeline and recover the valid blob with req/resp (randomly querying peers: any of our neighbours might have seen the valid blob)
* we regossip the valid blob (so as to ensure that the valid blob remains in the gossipsub caches)

Reply:
3a37c3c4978467f8909f66d8899b80b9e1373e72 - I've gone for the option that allows proposers to orphan evil block proposers that spread junk - to avoid network splits, we have to be careful to clear seen caches so that the valid block _can_ propagate - I *think* this should be enough to discourage anyone from trying, ie the network should _eventually_ settle on being able to gossip the block _or_ it will be orphaned depending on timing - both those outcomes are fine, even if they take up some bandwidth to get there.